test/mitmproxy/test_server.py

```diff
@@ -101,10 +101,16 @@ def test_sni(self):
         if not self.ssl:
             return
 
+        if hasattr(self, 'reverse') and self.reverse:
```

Comment:
Ha, that's definitely better :) I'm usually a Ruby guy, gotta brush up on my Python skills!